### ISSUE TITLE
Allow any ordering of xyz  kwarg

### ISSUE THREAD

**Issue by hmacdope** [MDAnalysis/transport-analysis#13]
It's mentioned above to include any ordering of `xyz`, which I think would be nice. A good way to implement that would be to convert the `dim_type` string to a set and then use sets as the keys here. That would give you flexibility without enumerating every option. e.g.

```python
dim_type = set(self.dim_type.split())

        keys = {
            {"x"}: [0],
            {"y"}: [1],
            {"z"}: [2],
            {"x", "y"}: [0, 1],
            {"x", "z"}: [0, 2],
            {"y", "z"}: [1, 2],
            {"x", "y", "z"}: [0, 1, 2],
        }

self._dim = keys[dim_type]

```

_Originally posted by @orionarcher in https://github.com/MDAnalysis/transport-analysis/pull/1#discussion_r1228445274_

**Comment by xhgchen** [comment]
Wait, isn't this a duplicate issue since #6 already exists?

**Comment by hmacdope** [comment]
Yes but wanted to capture @orionarcher solution. You can combine them and close one.
